osquery/tables/system/linux/apt_sources.cpp

```
@@ -198,6 +210,140 @@ static void genAptSource(const std::string& source,
+        return Status::failure("not a deb type repo");
+      }
+    }
+
+    if (key == "Enabled" && value != "on") {
+      return Status::failure("repo is not enabled");
+    }
+
+    if (key == "URIs") {
+      for (auto& uri : osquery::split(value)) {
+        if (uri.find("://") == std::string::npos) {
+          continue;
+        }
+        // Cannot have trailing slashes
+        while (uri.back() == '/') {
+          uri.pop_back();
+        }
+        uris.push_back(uri);
+      }
+    }
+
+    if (key == "Suites") {
+      for (const auto& suite : osquery::split(value)) {
+        suites.push_back(suite);
+      }
+    }
+  }
+
+  if (uris.size() == 0) {
+    return Status::failure("missing valid URIs");
+  }
+
+  if (suites.size() == 0) {
+    return Status::failure("missing Suites");
+  }
+
+  for (const auto& uri : uris) {
+    // All URLs should have protocol, checked earlier
+    auto proto_end = uri.find("://");
+    auto name = uri.substr(proto_end + 3);
+    for (const auto& suite : suites) {
+      auto suite_parts = osquery::split(suite, "/");
+      AptSource apt_source;
+
+      apt_source.base_uri = uri;
+      apt_source.name = name + ' ' + suite;
+      apt_source.cache_file.push_back(name);
+
+      if (suite_parts.size() == 1) {
+        apt_source.cache_file.push_back("dists");
+        apt_source.cache_file.push_back(suite);
+      } else {
+        apt_source.cache_file.insert(apt_source.cache_file.end(),
+                                     suite_parts.begin(),
+                                     suite_parts.end());
+      }
+
+      apt_sources.push_back(apt_source);
+    }
+  }
+
+  return Status::success();
+}
+
+void genDeb822Block(const std::string& source,
+                    const std::string& block,
+                    QueryData& results,
+                    Logger& logger) {
+  std::vector<AptSource> apt_sources;
+  parseDeb822Block(block, apt_sources);
```

Check for the returned `Status`?

---

I know if we want to return an error on invalid source, the original apt source doesn't check the return value either https://github.com/dantecatalfamo/osquery/blob/66accac3d3cea0be7f54ae69018410c94512c001/osquery/tables/system/linux/apt_sources.cpp#L209

---

I would return it and log the error (when in verbose mode) so it doesn't fail silently.

---

Am ok with keeping the consistency with `genAptUrl`&`aptSourceToRow`.

---

I can log it, but I don't think an invalid block should return early and prevent the rest of thee table from loading